base/docs/helpdb/Dates.jl

```diff
@@ -82,8 +128,9 @@ Dates.Date(dt::DateTime)
 """
     Date(dt::AbstractString, format::AbstractString; locale="english") -> Date
 
-Construct a `Date` type by parsing a `dt` date string following the pattern given in the
-`format` string. Follows the same conventions as `DateTime` above.
+Construct a `Date` object by parsing a `dt` date string following the pattern given in the
+`format` string. Follows the same conventions as
+`DateTime(::AbstractString, ::AbstractString)`.
```

Comment:
Should be possible to get that working I would think.

Currently the `?` mode behaviour tries to find the types by using  `typeof` on each argument, [here](https://github.com/JuliaLang/julia/blob/ec6f886a0d938046ae3adb47cf2b9cde6a5c1621/base/docs/utils.jl#L154-L158), and then when that "fails" fall back to using `Docs.signature` which just looks at the RHS of `arg :: TYPE`. `Base.gen_call_with_extracted_types` only fails on `:: Type` rather than `arg :: TYPE` though, which leads to the behaviour shown in those examples. This could definitely be made a bit more intelligent.